src/local/motion/NonRigidObjectTransformationUsingBSplines.cxx

```diff
@@ -230,11 +230,58 @@ set_deformation_field_from_file(DeformationFieldOnCartesianGrid<3,float>& deform
   return Succeeded::yes;
 }
 
+static
+Succeeded
+set_deformation_field_from_file(DeformationFieldOnCartesianGrid<3,float>& deformation_field,
+                const std::string& deformation_field_multicomponent_filename,
+                CartesianCoordinate3D<float>& grid_spacing,
```

Comment:
`const`, also next arg

Reply:
set `grid_spacing` and `origin` to const? They are set in this method. Do you want me to move them earlier in the argument list? i.e., 
```
DeformationFieldOnCartesianGrid<3,float>& deformation_field,
                CartesianCoordinate3D<float>& grid_spacing,
	        CartesianCoordinate3D<float>& origin,
                const std::string& deformation_field_multicomponent_filename)
```
and
```
DeformationFieldOnCartesianGrid<3,float>& deformation_field,
                CartesianCoordinate3D<float>& grid_spacing,
	        CartesianCoordinate3D<float>& origin,
    	        const std::string& deformation_field_from_file_x,
		const std::string& deformation_field_from_file_y,
		const std::string& deformation_field_from_file_z)
```

Reply:
if they're set, they indeed have to come first according to STIR conventions, so yes.

probably document in the declaration.